client/src/connection_cache.rs

```diff
@@ -22,26 +22,14 @@ enum Connection {
 }
 
 struct ConnMap {
-    // Keeps track of the connection associated with an addr and the last time it was used
-    map: HashMap<SocketAddr, (Connection, u64)>,
-    // Helps to find the least recently used connection. The search and inserts are O(log(n))
-    // but since we're bounding the size of the collections, this should be constant
-    // (and hopefully negligible) time. In theory, we can do this in constant time
-    // with a queue implemented as a doubly-linked list (and all the
-    // HashMap entries holding a "pointer" to the corresponding linked-list node),
-    // so we can push, pop and bump a used connection back to the end of the queue in O(1) time, but
-    // that seems non-"Rust-y" and low bang/buck. This is still pretty terrible though...
-    last_used_times: BTreeMap<u64, SocketAddr>,
-    ticks: u64,
+    map: LruCache<SocketAddr, Connection>,
     use_quic: bool,
 }
 
 impl ConnMap {
     pub fn new() -> Self {
         Self {
-            map: HashMap::new(),
-            last_used_times: BTreeMap::new(),
-            ticks: 0,
+            map: LruCache::new(MAX_CONNECTIONS),
```

Comment:
Is 100 enough? How many do we need realistically in production?

Reply:
In addition not related to this PR -- I think the singleton of the connection cache need to be reconsidered. For example, if we want multiple connections for the same address, it would be impossible to do. One way to solve that is leave the client to create its own cache.

Reply:
I think we should address the value of MAX_CONNECTIONS in another PR - there's actually already another PR to change it anyway. Likewise, I think it's intentional to have only 1 connection per address, but either way I think it's more appropriate to address that in another PR.